src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -513,8 +513,9 @@ private function renderCell(array $row, int $column, string $cellFormat): string
         }
 
         // str_pad won't work properly with multi-byte strings, we need to fix the padding
-        if (false !== $encoding = mb_detect_encoding($cell, null, true)) {
-            $width += \strlen($cell) - mb_strwidth($cell, $encoding);
+        $cellstr = (string) $cell;
```

Comment:
$call as name would save renaming below,thus save merge conflicts, please keep the name

Reply:
I had it done if `$cell` wasn't used below :
`$cell instanceof TableSeparator`

Do you prefer that I store `$cell instanceof TableSeparator` in a variale, so I can rewrite this as

	$isTableSeparator = ($cell instanceof TableSeparator)

	$cell = (string) $cell;
	if (false !== $encoding = mb_detect_encoding($cell, null, true)) {
		$width += \strlen($cell) - mb_strwidth($cell, $encoding);
	}

	$style = $this->getColumnStyle($column);

	if ($isTableSeparator) {
		return sprintf($style->getBorderFormat(), str_repeat($style->getBorderChars()[2], $width));
	}

Reply:
Oh, I missed this.
But: is this needed at all? $cell cannot be null here.

Reply:
I must have missed something, I wonder how you can be sure ?
In fact, addRow is public, and can take an array of anything, since nothing enforces the type of the elements. So, IMHO, in renderRow, you can't assume anything on the type of elements of row. And, therefore, you can't assume anything on the type of $cell.

BTW, even if it was right, the aim of this PR is to avoid errors when upgrading to PHP8.1 (or to the usage of the polyfill) by transtyping _before_ the call to the mb_ function, rather than relying on the implict cast done until php8.0

I originally had the problem with a null value, you're right, but the problem is not related with the value being null, it is related with the value being anything else than a string.
Arguments of mb_* functions _must_ be a string (in PHP8.1, and therefore it has been coded in the polyfill). Passing them an object won't be good.

So :
- In any way, it can be an object (since code is testing for it)
- It comes from a public method, with no type checking - not even in phpdoc -, so I don't see how you can assume anything about its type.

(All that being said, I am perfectly conscious about the work you provide, and the very good knowledge of this code you have. That's why you're reviewers and I'm just a first-time request-puller :) )

Reply:
Passing them a Stringable object will continue to work, isn't it?
That's how this piece of code is designed to me, and it's fine as is.

Reply:
ADAIK, the prototype of the mb_ functions is marked (in the php documentation) as "string", not "string|Stringable"

And if I undesrtand correctly https://php.watch/versions/8.0/stringable, this has been implemented this way (I mean, a type hinting of "string" doesn't seem to mean "string|Stringable")

But I'm not really sure, it would require the advice of an expert.

Well, wait a minute... I just saw the author of he RFC for Stringable is nobody else than you...
So it seems you're an expert :)

It's up to you to see : if I'm wrong, and mb_ functions accept string|Stringable although their prototype says "string", then you're right, and I can remove the cast.

Reply:
> mb_ functions accept string|Stringable although their prototype says "string"

Yes, that works, because we don't use strict mode (and we don't want to).

Reply:
> mb_ functions accept string|Stringable although their prototype says "string"

It does, unless you enable `strict_types`. 🙃

Reply:
Okay, then.
Fixed